bench/BenchAll.hs

```diff
@@ -481,6 +482,18 @@ main = do
       [ bench "lazy"   $ nf L8.unlines (map (L8.pack . show) intData)
       , bench "strict" $ nf S8.unlines (map (S8.pack . show) intData)
       ]
+    , bgroup "pack"
+      [ bench "not fused" $ nf S.pack (replicate nRepl 0)
+      , bench "fused" $ nf (S.pack . replicate nRepl) 0
```

Comment:
Why is there a difference in fusion between these two lines. I think this should probably be fixed. Maybe `packBytes` and `packChars` need to be eta-reduced?!

Reply:
It is because the `nf` function reapplies its second argument many times to its first argument. If you put the list generation in the second argument, then the lists that are generated are shared between calls to the first argument and hence they cannot be fused. If you put the list generation code in the first argument then a new list will be generated for every call and each of those lists can be fused.

Reply:
Thanks for the explanation!